dtfs-central-api/api-tests/v1/gef-deals/add-min-activities.api-test.js

```diff
@@ -0,0 +1,413 @@
+const { format, fromUnixTime } = require('date-fns');
+
+const {
+  ukefSubmissionPortalActivity,
+  facilityChangePortalActivity,
+  portalActivityGenerator,
+  getUserInfo,
+  updateChangedToIssued,
+} = require('../../../src/v1/controllers/portal/gef-deal/add-min-activities.controller');
+
+const collectionName = 'facilities';
+const applicationCollectionName = 'deals';
+
+const db = require('../../../src/drivers/db-client');
+
+const mockApplications = require('../../mocks/gef/gef-applications');
+const { mockFacilities } = require('../../mocks/gef/gef-facilities');
+const CONSTANTS = require('../../../src/constants');
+const { PORTAL_ACTIVITY_LABEL, PORTAL_ACTIVITY_TYPE } = require('../../../src/constants/activityConstants');
+
+const MOCK_APPLICATION = mockApplications[0];
+const MOCK_APPLICATION_FACILITIES = mockApplications[15];
+
+const wipeDB = require('../../wipeDB');
+
+const testUserCache = require('../../mocks/test-users/api-test-users');
+const app = require('../../../src/createApp');
+
+const { as } = require('../../api')(app);
+
+const baseUrl = '/v1/portal/gef/facilities';
+const applicationBaseUrl = '/v1/portal/gef/deals';
+
+describe('submissionPortalActivity()', () => {
+  it('should return a populated array with submission activity object and MIA', async () => {
+    await wipeDB.wipe([collectionName]);
+    await wipeDB.wipe([applicationCollectionName]);
+    /*
+   As _id's can change for checker, need to access db and find a checker
+   These details then added to the MOCK_APPLICATION
+   */
+    const userCollection = await db.getCollection('users');
+    // finds someone with role checker only
+    const checker = await userCollection.findOne({ roles: ['checker'] });
```

Comment:
is there a way to get these details from a mock file? or an API call? I'd like to avoid calling mongdb from the api-test directly

Reply:
the code works by making a mongo call to find the user (to populate the by field). I would have to insert into mongodb first and then mock it, so works out the same?